### PR TITLE
Support TorchVision v2 transforms with ImageInfo

### PR DESCRIPTION
**Pull request by vinnamkim**
### Summary

- It is a left-over from https://github.com/openvinotoolkit/training_extensions/pull/2709
- Change `ImageInfo` from `dataclass` to `TVTensor` to register it to TorchVision v2 geometry and normalize transform kernels: `Resize`, `RandomResize`, `RandomCrop`, `RandomResizedCrop`, `CenterCrop`, `Pad`, and `Normalize`. 

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by sungmanc — DISMISSED** [review]

File: src/otx/core/data/dataset/action_detection.py [hunk 1]
```diff
@@ -45,7 +45,6 @@ def _get_item_impl(self, idx: int) -> ActionDetDataEntity | None:
                 img_idx=idx,
                 img_shape=img_shape,
                 ori_shape=img_shape,
-                pad_shape=img_shape,
                 scale_factor=(1.0, 1.0),
```
Comment (jaegukhyun) [inline src/otx/core/data/dataset/action_detection.py]: Does action detection require scale_factor?

File: src/otx/core/data/entity/base.py [hunk 2]
```diff
@@ -11,29 +11,249 @@
...
-class ImageInfo:
+class ImageInfo(tv_tensors.TVTensor):
     """Meta info for image.
 
-    :param img_id: Image id
-    :param img_shape: Image shape after preprocessing
-    :param ori_shape: Image shape right after loading it
-    :param pad_shape: Image shape before padding
-    :param scale_factor: Scale factor if the image is rescaled during preprocessing
+    Attributes:
```
Comment (jaegukhyun) [inline src/otx/core/data/entity/base.py:26]:
```suggestion
    Args:
```

Comment (vinnamkim) [reply src/otx/core/data/entity/base.py:26]: It was intended. I'd like to notate them as attributes in this class.

File: src/otx/core/data/dataset/action_detection.py (hunk 1 above)
Comment (vinnamkim) [reply src/otx/core/data/dataset/action_detection.py]: Why not? Is the action detection has images?

Comment (jaegukhyun) [reply src/otx/core/data/dataset/action_detection.py]: I mean all the other algo removed scale_factor attribute, but action detection still takes scale_factor as an input.

Comment (vinnamkim) [reply src/otx/core/data/dataset/action_detection.py]: Fixed in 0b833ae4920894981f161eacaac0c791acca9b27

**Review by jaegukhyun — DISMISSED** [review]

**Review by sungmanc — DISMISSED** [review]

**Review by harimkang — DISMISSED** [review]

**Review by harimkang — APPROVED** [review]

**Review by jaegukhyun — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Pull request merged**